### DISCUSSION TITLE
Cors error on the network

### DISCUSSION THREAD

**Discussion by HUGOALMEIDARAMOS** [adonisjs/core#2723] [Help]
Good Morning

sorry for my english, i'm brazilian and i don't speak the language.

I'm having problems with cors.

On the server itself, opening the browser (http://localhost:8081) works, I can access the system, but when I access it through a network machine (http://ip_da_maquina:8081) I can't.

Here is my cors.js file

--------------------------------------------------------------------------------------------------------------------------------------------------------------

'use strict'

module.exports = {

  enabled : true ,

 
  

  origin: [
    'http://localhost',
    'http://localhost:8081',
    '*'
    ],

  methods: ['GET', 'PUT', 'PATCH', 'POST', 'DELETE'],

  headers: true,

  
  exposeHeaders: [
    'cache-control',
    'content-language',
    'content-type',
    'expires',
    'last-modified',
    'pragma'
  ],

  credentials: true,

  
  maxAge: 90
}

**Comment by d4n1b** [comment]
Hey @HUGOALMEIDARAMOS,

Try adding 'OPTIONS' to the array of methods.

I hope it helps.

**Reply by HUGOALMEIDARAMOS** [reply]
I'm new to the backend, how could I add these options. on the server itself, opening a browser, I can access it, but from other machines on the network, an error occurs, as shown in the image below

**Reply by HUGOALMEIDARAMOS** [reply]
This error occurs when I try to access from another machine on the network.

![WhatsApp Image 2021-06-02 at 13 49 31 (1)](https://user-images.githubusercontent.com/16476471/120521170-02e02500-c3ab-11eb-9c24-a1134b7e2c02.jpeg)
![WhatsApp Image 2021-06-02 at 13 49 31](https://user-images.githubusercontent.com/16476471/120521175-04115200-c3ab-11eb-8e0f-a05d842f0b0e.jpeg)

**Reply by d4n1b** [reply]
Oh, sorry. I meant on the code you shared in cors.js file:

From
`methods: ['GET', 'PUT', 'PATCH', 'POST', 'DELETE'],`

To
`methods: ['GET', 'PUT', 'PATCH', 'POST', 'DELETE', 'OPTIONS'],`

**Reply by HUGOALMEIDARAMOS** [reply]
keeps giving refused  :(

**Reply by d4n1b** [reply]
Hi @HUGOALMEIDARAMOS ,

If you're in development mode, you can just use `origin: true,` otherwise you need to whitelist the IPs you want to approve.

The issue is that none of the origins in your config are listing the other computers' IP

```ts
origin: [
  'http://localhost',
  'http://localhost:8081',
  '*'
],
```

I hope it helps.
